src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -816,29 +814,28 @@ public unsafe void CopyTo(Array array, int index)
                 Vector128<byte> lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
                 Vector128<byte> upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
 
-                if (Avx2.IsSupported)
+                if (Vector256.IsHardwareAccelerated)
                 {
                     Vector256<byte> shuffleMask = Vector256.Create(lowerShuffleMask_CopyToBoolArray, upperShuffleMask_CopyToBoolArray);
                     Vector256<byte> bitMask = Vector256.Create(0x80402010_08040201).AsByte();
                     Vector256<byte> ones = Vector256.Create((byte)1);
 
-                    fixed (bool* destination = &boolArray[index])
+                    ref byte destination = ref Unsafe.As<bool, byte>(ref Unsafe.Add<bool>(ref MemoryMarshal.GetArrayDataReference<bool>(boolArray), index));
+
+                    for (; (i + Vector256ByteCount) <= (uint)m_length; i += Vector256ByteCount)
                     {
-                        for (; (i + Vector256ByteCount) <= (uint)m_length; i += Vector256ByteCount)
-                        {
-                            int bits = m_array[i / (uint)BitsPerInt32];
-                            Vector256<int> scalar = Vector256.Create(bits);
-                            Vector256<byte> shuffled = Avx2.Shuffle(scalar.AsByte(), shuffleMask);
-                            Vector256<byte> extracted = Avx2.And(shuffled, bitMask);
-
-                            // The extracted bits can be anywhere between 0 and 255, so we normalise the value to either 0 or 1
-                            // to ensure compatibility with "C# bool" (0 for false, 1 for true, rest undefined)
-                            Vector256<byte> normalized = Avx2.Min(extracted, ones);
-                            Avx.Store((byte*)destination + i, normalized);
-                        }
+                        int bits = m_array[i / (uint)BitsPerInt32];
+                        Vector256<int> scalar = Vector256.Create(bits);
```

Comment:
Is the JIT correctly treating this as a `vpbroadcastd dst, [addr]`?

Is it intentional that we don't cache `m_array` in a local to avoid repeated null and range checks each iteration?